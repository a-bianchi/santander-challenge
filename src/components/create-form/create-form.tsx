import { makeStyles, Theme } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import { useTranslation } from 'react-i18next';
import { useFormik } from 'formik';
import * as Yup from 'yup';
import { TextField } from '@material-ui/core';

const useStyles = makeStyles((theme: Theme) => ({
    root: {
        flexGrow: 1,
    },
    form: {
        width: '100%', // Fix IE 11 issue.
        marginTop: theme.spacing(1),
    },
    submit: {
        margin: theme.spacing(3, 0, 2),
    },
    error: {
        color: theme.palette.error.dark,
    },
}));

interface Props {
    create: any;
}

export const CreateForm = ({ create }: Props): JSX.Element => {
    const classes = useStyles();
    const { t } = useTranslation();

    const formValidationSchema = Yup.object().shape({
        title: Yup.string().required(t('createForm.errorNamePresence')),
        date: Yup.string().required(t('createForm.errorDatePresence')),
        beers: Yup.number().required(t('createForm.errorBeersQuantityPresence')),
        people: Yup.number().required(t('createForm.errorPeoplePresence')),
    });

    const formik = useFormik({
        initialValues: {
            title: '',
            date: '',
            temperature: 0,
            people: 0,
            beers: 0,
        },
        validationSchema: formValidationSchema,
        onSubmit: (values) => {
            console.log(JSON.stringify(values));
            create({ ...values, status: 'pending' });
        },
    });

    return (
        <div className={classes.root}>
            <form data-test="CreateForm" className={classes.form} onSubmit={formik.handleSubmit}>
                <TextField
                    data-test="inputCreateName"
                    variant="outlined"
                    margin="normal"
                    required
                    fullWidth
                    id="title"
                    label={t('createForm.title')}
                    name="title"
                    autoFocus
                    onChange={formik.handleChange}
                    value={formik.values.title}
                    error={formik.errors.title ? true : false}
                />
                {formik.errors.title && formik.touched.title ? (
                    <div data-test="errorMessageCreateTitle" className={classes.error}>
                        {formik.errors.title}
                    </div>
                ) : null}
                <TextField
                    data-test="inputCreateDate"
                    variant="outlined"
                    margin="normal"
                    required
                    fullWidth
                    name="date"
                    type="date"
                    id="date"
                    onChange={formik.handleChange}
                    value={formik.values.date}
                    error={formik.errors.date ? true : false}
                />
                {formik.errors.date && formik.touched.date ? (
                    <div data-test="errorMessageCreateDate" className={classes.error}>
                        {formik.errors.date}
                    </div>
                ) : null}
                <TextField
                    data-test="inputCreateTemperature"
                    variant="outlined"
                    margin="normal"
                    fullWidth
                    disabled={true}
                    name="temperature"
                    label={t('createForm.temperature')}
                    type="number"
                    id="temperature"
                    onChange={formik.handleChange}
                    value={formik.values.temperature}
                />
                <TextField
                    data-test="inputCreatePeople"
                    variant="outlined"
                    margin="normal"
                    required
                    fullWidth
                    name="people"
                    label={t('createForm.people')}
                    type="number"
                    id="people"
                    onChange={formik.handleChange}
                    value={formik.values.people}
                    error={formik.errors.people ? true : false}
                />
                {formik.errors.people && formik.touched.people ? (
                    <div data-test="errorMessageCreatePeople" className={classes.error}>
                        {formik.errors.people}
                    </div>
                ) : null}
                <TextField
                    data-test="inputCreateBeers"
                    variant="outlined"
                    margin="normal"
                    required
                    fullWidth
                    name="beers"
                    label={t('createForm.beers')}
                    type="number"
                    id="beers"
                    onChange={formik.handleChange}
                    value={formik.values.beers}
                    error={formik.errors.beers ? true : false}
                />
                {formik.errors.beers && formik.touched.beers ? (
                    <div data-test="errorMessageCreateBeers" className={classes.error}>
                        {formik.errors.beers}
                    </div>
                ) : null}
                <Button
                    data-test="SubmitCreateForm"
                    type="submit"
                    fullWidth
                    variant="contained"
                    color="primary"
                    className={classes.submit}
                >
                    {t('createForm.submit')}
                </Button>
            </form>
        </div>
    );
};
