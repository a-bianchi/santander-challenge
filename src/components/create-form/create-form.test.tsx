/* global describe expect it */
import { shallow } from 'enzyme';
import { CreateForm } from './create-form';

describe('Create Form', () => {
    it('should be defined', () => {
        expect(CreateForm).toBeDefined();
    });
    it('should render correctly', () => {
        const tree = shallow(
            <CreateForm
                create={() => {
                    console.log('test');
                }}
            />,
        );
        expect(tree.find('.CreateForm')).toBeDefined();
        expect(tree.find('.inputCreateName')).toBeDefined();
        expect(tree.find('.inputCreateDate')).toBeDefined();
        expect(tree.find('.inputCreateTemperature')).toBeDefined();
        expect(tree.find('.inputCreatePeople')).toBeDefined();
        expect(tree.find('.inputCreateBeers')).toBeDefined();
        expect(tree.find('.SubmitCreateForm')).toBeDefined();
        expect(tree).toMatchSnapshot();
    });
});
