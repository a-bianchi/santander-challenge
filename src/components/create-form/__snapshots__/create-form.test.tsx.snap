// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Form should render correctly 1`] = `ShallowWrapper {}`;
